packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,344 @@
+                take: limit,
+              } as FindProductConfig)
+          }
+
+          products.forEach((product: Product) => {
+            const lines = this.buildProductVariantLines(product)
+            lines.forEach((line) => writeStream.write(line))
+          })
+
+          advancementCount += products.length
+          offset += products.length
+          products.length = 0
+
+          batchJob = (await this.batchJobService_
+            .withTransaction(transactionManager)
+            .update(batchJobId, {
+              result: {
+                file_key: fileKey,
+                count: productCount,
+                advancement_count: advancementCount,
+                progress: advancementCount / productCount,
+              },
+            })) as ProductExportBatchJob
+
+          if (batchJob.status === BatchJobStatus.CANCELED) {
+            writeStream.end()
+
+            await this.fileService_
+              .withTransaction(transactionManager)
+              .delete({ key: fileKey })
+            return
+          }
+        }
+
+        writeStream.end()
+
+        return await promise
+      },
+      "REPEATABLE READ",
+      async (err) =>
+        this.handleProcessingError(batchJobId, err, {
+          count: productCount,
+          advancement_count: advancementCount,
+          progress: advancementCount / productCount,
+        })
+    )
+  }
+
+  public async buildHeader(batchJob: ProductExportBatchJob): Promise<string> {
+    const {
+      dynamicImageColumnCount,
+      dynamicMoneyAmountColumnCount,
+      dynamicOptionColumnCount,
+    } = batchJob.context.shape
+
+    this.appendOptionsDescriptors(dynamicOptionColumnCount)
+    this.appendMoneyAmountDescriptors(dynamicMoneyAmountColumnCount)
+    this.appendImagesDescriptors(dynamicImageColumnCount)
+
+    return [...this.columnDescriptors.keys(), this.NEWLINE_].join(
+      this.DELIMITER_
+    )
+  }
+
+  private appendImagesDescriptors(maxImagesCount: number): void {
+    for (let i = 0; i < maxImagesCount; ++i) {
+      this.columnDescriptors.set(`Image ${i + 1} Url`, {
+        accessor: (product: Product) => product?.images[i]?.url ?? "",
+        entityName: "product",
+      })
+    }
+  }
+
+  private appendOptionsDescriptors(maxOptionsCount: number): void {
+    for (let i = 0; i < maxOptionsCount; ++i) {
+      this.columnDescriptors
+        .set(`Option ${i + 1} Name`, {
+          accessor: (productOption: Product) =>
+            productOption?.options[i]?.title ?? "",
+          entityName: "product",
+        })
+        .set(`Option ${i + 1} Value`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.options[i]?.value ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private appendMoneyAmountDescriptors(maxMoneyAmountCount: number): void {
+    for (let i = 0; i < maxMoneyAmountCount; ++i) {
+      this.columnDescriptors
+        .set(`Price ${i + 1} Currency code`, {
```

Comment:
This should be Price DKK, Price EUR, etc. `"Currency code"` is just the placeholder.

Reply:
But isn’t it better to have it that way so that the values are part of the lines and not in the column name?

Reply:
<img width="294" alt="Screenshot 2022-06-21 at 16 15 27" src="https://user-images.githubusercontent.com/25098370/174821741-6f8ccc75-ce9e-427f-8059-9808649d76f9.png">
from the column here, the DKK, EUR correspond to the amount i suppose, then the price currency_code would also be DKK, EUR and then what the price France would contains?

Reply:
Great! Basically, we don't want to put the currency code into the cells because then for example excel won't treat it as a number and you can't do excel operations on it e.g. sum, etc.

Reply:
I mean in the currency cell as it was 😂